packages/hint/src/lib/cli/analyze.ts

```diff
@@ -373,31 +374,75 @@ export default async (actions: CLIOptions): Promise<boolean> => {
         }
     };
 
-    engine.on('print', print);
+    const scan = async (eng: Engine, target: URL) => {
+        let code = 0;
+        let newUrl: URL | null = null;
+        const spinner: ORA = ora({ spinner: 'line' });
+
+        initSpinner(spinner, eng, actions.debug);
 
-    for (const target of targets) {
         try {
             const scanStart = Date.now();
-            const results: Array<Problem> = await engine.executeOn(target);
+            const results: Array<Problem> = await eng.executeOn(target);
             const scanEnd = Date.now();
 
             if (hasError(results)) {
-                exitCode = 1;
+                code = 1;
             }
 
-            endSpinner(exitCode ? 'fail' : 'succeed');
+            endSpinner(spinner, code ? 'fail' : 'succeed', actions.debug);
 
             await askForTelemetryConfirmation(config);
             await print(results, target.href, scanEnd - scanStart, new Date(scanStart).toISOString());
         } catch (e) {
-            exitCode = 1;
-            endSpinner('fail');
+            code = e.name === 'RedirectError' ? 2 : 1;
+
+            if (code === 2) {
+                newUrl = e.newUrl;
+            }
+            endSpinner(spinner, 'fail', actions.debug);
             debug(`Failed to analyze: ${target.href}`);
             debug(e);
         }
-    }
 
-    await engine.close();
+        await eng.close();
+
+        return {
+            code,
+            newUrl
+        };
+    };
+
+    const createEngine = () => {
+        const eng = new Engine(config, resources);
+
+        eng.on('print', print);
+
+        return eng;
+    };
+
+    for (const target of targets) {
+        let code: number = 0;
+        let result: { code: number, newUrl: URL | null };
+
+        engine = createEngine();
+        result = await scan(engine!, target);
+
+        code = result.code;
+
+        while (code === 2) {
+            engine = createEngine();
+            logger.log(`Redirect detected, analyzing: ${result.newUrl!.href}`);
```

Comment:
What about the following:

```suggestion
            logger.log(`Client side redirect detected, analyzing: ${result.newUrl!.href}`);
```